homeassistant/components/obihai/sensor.py

```diff
@@ -87,26 +83,40 @@ def icon(self) -> str:
 
     def update(self) -> None:
         """Update the sensor."""
-        if not self._pyobihai.check_account():
-            self._attr_native_value = None
-            self._attr_available = False
-            return
 
-        services = self._pyobihai.get_state()
+        LOGGER.debug("Running update on %s", self._service_name)
+        try:
+            # port connection, and last caller info
+            if "Caller Info" in self._service_name or "Port" in self._service_name:
+                services = self._pyobihai.get_line_state()
 
-        if self._service_name in services:
-            self._attr_native_value = services.get(self._service_name)
+                if services is not None and self._service_name in services:
+                    self._attr_native_value = services.get(self._service_name)
+            elif self._service_name == "Call Direction":
+                call_direction = self._pyobihai.get_call_direction()
 
-        services = self._pyobihai.get_line_state()
+                if self._service_name in call_direction:
+                    self._attr_native_value = call_direction.get(self._service_name)
+            else:  # SIP Profile service sensors, phone sensor, and last reboot
+                services = self._pyobihai.get_state()
 
-        if services is not None and self._service_name in services:
-            self._attr_native_value = services.get(self._service_name)
+                if self._service_name in services:
+                    self._attr_native_value = services.get(self._service_name)
 
-        call_direction = self._pyobihai.get_call_direction()
+            if not self.requester.available:
+                self.requester.available = True
+                LOGGER.warning("Connection restored")
```

Comment:
Decrease this to info
```suggestion
                LOGGER.info("Connection restored")
```